src/Npgsql/NpgsqlCommand.cs

```diff
@@ -605,48 +605,46 @@ async Task PrepareLong()
 
                     // Loop over statements, skipping those that are already prepared (because they were persisted)
                     var isFirst = true;
-                    foreach (var statement in _statements.Where(s =>
-                        s.PreparedStatement?.State == PreparedState.BeingPrepared))
-                    {
-                        var pStatement = statement.PreparedStatement;
-                        Debug.Assert(pStatement != null);
-                        Debug.Assert(pStatement.Description == null);
-                        if (pStatement.StatementBeingReplaced != null)
+                    foreach (var statement in _statements)
+                        if (statement.PreparedStatement?.State == PreparedState.BeingPrepared)
```

Comment:
Is all this to save the single allocation in LINQ Where? PrepareLong is really not perf-sensitive, so I think it's OK to use LINQ here.

If you want to keep this, then please reverse the logic - if the state is not BeingPrepared then continue, and have curlies around the foreach block. I like to omit curlies for one-liners but here it's a bit much :)

Reply:
Not for perf. See the comment below.